soc: mimxrt1180: Add USB Device support

Tim-Wang38 opened the pull request:
This was tested on the MIMXRT1180 EVK board

Tim-Wang38 committed:
soc: mimxrt1180: Add USB Device support

This was tested on the MIMXRT1180 EVK board

Signed-off-by: Tim Wang <tim.wang@nxp.com>

Tim-Wang38 commented:
wrong PR to zephyr

dleach02 commented:
@Tim-Wang38 Please change the commit message to use "USB Device" instead of USB2. Otherwise, looks good.

Tim-Wang38 commented:
Hi @dleach02 , I have changed the commit content and also deleted the blank row on the mimxrt1180_evk_mimxrt1189_cm7.dts file, please help to review. Thanks.

dleach02 commented:
@Tim-Wang38, I rebased this PR to pull in https://github.com/zephyrproject-rtos/zephyr/commit/06f42f6bfa420b6914ead8cdc1de9add6491e1b9 fix which was blocking CI

Tim-Wang38 commented:
@dleach02 , thanks a lot for your help!
